src/internal/api/authenticator_test.go

```diff
@@ -261,127 +277,78 @@ func TestLogout(t *testing.T) {
-			name:       "should fire first backoff when encountered problem for the first time",
-			errIn:      errors.New("error"),
-			backoffCfg: config.BackoffCfg{MaxAttempts: 2, Length: "0"},
-			auth: authenticator{
-				backoffCfg: backoffCfg{
-					status:   statusWorkingProperly,
-					attempts: 0,
-				},
-			},
-			expectedStatus: statusWaitingToReconnect,
-			errorContains:  "failed to refresh token",
-		},
-		{
-			name:       "should return error when invalid status is encountered",
-			errIn:      errors.New("error"),
-			backoffCfg: config.BackoffCfg{MaxAttempts: 2, Length: "0"},
-			auth: authenticator{
-				backoffCfg: backoffCfg{
-					status:   statusConnectionFailed,
-					attempts: 0,
-				},
-			},
-			expectedStatus: statusConnectionFailed,
-			errorContains:  "invalid auth status",
-		},
+	backoff := backoff.NewStateful(
+		5*time.Second,
+		1*time.Second,
+		1*time.Second,
+		1,
+		1,
+	)
+
+	handler := func(w http.ResponseWriter, _ *http.Request) {
+		data := fmt.Sprintf(`{"accessToken":"%s","refreshToken":"%s"}`,
+			accessToken,
+			refreshToken,
+		)
+
+		w.WriteHeader(http.StatusNotFound)
+
+		_, _ = w.Write([]byte(data))
 	}
+	server := httptest.NewServer(http.HandlerFunc(handler))
 
-	for _, val := range testCases {
-		v := val
-		t.Run(v.name, func(t *testing.T) {
-			t.Parallel()
+	defer server.Close()
 
-			// mock cfgSvc
-			cfg := config.Config{Credentials: config.Credentials{}, Backoff: v.backoffCfg}
-			storage := mockedstorage.Storage[*config.Config]{}
-			storage.On("Model").Return(&cfg)
-			storage.On("Save").Return(v.saveError)
+	cfg := config.Config{Credentials: config.Credentials{
+		AccessToken:           accessToken,
+		RefreshToken:          refreshToken,
+		RefreshTokenExpiresAt: time.Now().Add(time.Hour),
+		AccessTokenExpiresAt:  time.Now(),
+	}, Backoff: config.BackoffCfg{Length: "0"}}
 
-			notificationManager := &NotificationMock{}
-			notificationManager.On("Event", &notification.Event{EventName: notificationEaseeStatusOffline}).Return(nil)
-
-			cfgDir := path.Join("./../../testdata/testing/", "")
-			mqttCfg := config.New(cfgDir)
-			v.auth.mqtt = fimpgo.NewMqttTransport(
-				mqttCfg.MQTTServerURI,
-				mqttCfg.MQTTClientIDPrefix,
-				mqttCfg.MQTTUsername,
-				mqttCfg.MQTTPassword,
-				true,
-				1,
-				1)
-
-			v.auth.notificationManager = notificationManager
-			v.auth.cfgSvc = config.NewService(&storage)
-			err := v.auth.handleFailedRefreshToken(v.errIn)
-			assert.Equal(t, v.expectedStatus, v.auth.status)
-			assert.Contains(t, err.Error(), v.errorContains)
-			notificationManager.AssertNumberOfCalls(t, "Event", v.notificationManagerCalled)
-		})
+	storage := mockedstorage.Storage[*config.Config]{}
+	storage.On("Model").Return(&cfg)
+	storage.On("Save").Return("ok")
+
+	cfgSrv := config.NewConfigServiceWithStorage(&storage)
+
+	notificationManager := &NotificationMock{}
+
+	httpClient := &http.Client{Timeout: 3 * time.Second}
+	client := NewHTTPClient(cfgSrv, httpClient, server.URL)
```

Comment:
HTTP client should be mocked, as it already has a separate set of tests. It's gonna make the setup simpler here as well.